backend/src/ee/services/dynamic-secret/providers/elastic-search.ts

```diff
@@ -0,0 +1,176 @@
+import { Client as ElasticCacheClient } from "@elastic/elasticsearch";
+import handlebars from "handlebars";
+import { customAlphabet } from "nanoid";
+import { z } from "zod";
+
+import { getConfig } from "@app/lib/config/env";
+import { BadRequestError } from "@app/lib/errors";
+import { alphaNumericNanoId } from "@app/lib/nanoid";
+
+import { DynamicSecretElasticSearchSchema, TDynamicProviderFns } from "./models";
+
+const generatePassword = () => {
+  const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-_.~!*$#";
+  return customAlphabet(charset, 64)();
+};
+
+const generateUsername = () => {
+  return alphaNumericNanoId(32);
+};
+
+const parseJsonStatement = (str: string) => {
+  try {
+    return JSON.parse(str) as object;
+  } catch {
+    throw new BadRequestError({ message: "Failed to parse ElasticSearch statements" });
+  }
+};
+
+const CreateElasticSearchUserSchema = z
```

Comment:
Let's avoid this. The creation, revocation string statement is majorly used for db query as the statement would be complex to provide as option. 

But for JSON structured API lets keep it as in API level itself like the atlas provider. Because the issue is we want better ux for end user. Putting a JSON like that would confuse the user to understand how it's working. 

Let's move it inside the original the schema and use that.

Reply:
This would also allow us to be me more controlled